src/harness/reference_models/geo/refractivity.py

```diff
@@ -0,0 +1,105 @@
+#    Copyright 2017 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""Returns the ITU surface refractivity for a specified lat/lon.
+
+Standalone execution:
+  python refractivity.py lat lng
+"""
+
+import logging
+import math
+import numpy as np
+import os
+import sys
+
+from reference_models.geo import CONFIG
+
+
+class RefractivityIndexer:
+  """RefractivityIndexer class to retrieve refractivity for given world location.
+
+  It retrieves the nearest four refractivity points from ITU data file with
+  1.5 degree grid, and interpolates bilinearly between those points.
+
+  Typical usage:
+    refractor = RefractivityIndexer()
+    r = refractor.Refractivity(19.66, -155.55)
+  """
+  def __init__(self, datafile_or_dir=None):
+    self.ConfigureDataFile(datafile_or_dir, do_load=False)
+    self._num_lat_rows = 121  # Number of latitude rows in the data file
+    self._num_lon_cols = 241  # Number of longitude columns in the data file
+    self._lat_start = 90.0  # Latitude corresponding to first row of file (deg)
+    self._lon_start = 0.0  # Longitude corresponding to first column of file (deg)
+    self._delta_lat = self._delta_lon = 1.5  # Spacing between lat/lon rows/columns (deg)
+
+  def ConfigureDataFile(self, datafile_or_dir, do_load=True):
+    """Configure the refractivity data file.
+
+    Inputs:
+      datafile_or_dir: the data path or directory.
+         If directory, then the datafile is the standard 'n050.txt'.
+         If None, then use the standard database location from CONFIG.py.
+      do_load: if set (default), load the data, otherwise do lazy loading.
+    """
+    self._datafile = datafile_or_dir
+    if self._datafile is None:
+      self._datafile = os.path.join(CONFIG.GetItuDir(), 'n050.txt')
+    elif os.path.isdir(self._datafile):
+      self._datafile = os.path.join(self._datafile, 'n050.txt')
+    self._data = None
+    if do_load:
+      self._data = np.loadtxt(self._datafile)
+
+  def Refractivity(self, lat, lon):
+    """Returns ITU refractivity for the specified lat/lon.
+
+    Inputs:
+      lat, lon : the coordinates of a point.
+
+    Returns:
+      the sea level refractivity on that point.
+    """
+    if self._data is None:
+      self._data = np.loadtxt(self._datafile)
+      logging.info('Loaded refractivity data from %s' % self._datafile)
+
+    if lon < 0:
+      lon = lon + 360.0
```

Comment:
comment from Commscope

The lines (79-80) are not needed. Longitude falls between -180 and 180 for the data file (Spectrum-Access-System/src/harness/reference_models/geo/testdata/itu/n050.txt ).

Reply:
see previous comment

Reply:
comment also withdrawn, please see above